### PR TITLE
Add clj-kondo

### PR DESCRIPTION
**Pull request by solar05**
Before submitting a PR make sure the following things have been done:

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [ ] You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [ ] You've updated the [changelog](../CHANGELOG.md)(that only applies to user-visible changes)

Thanks!

*If you're just starting out to hack on nREPL you might find this [section of its
manual][1] extremely useful.*

[1]: https:/nrepl.org/nrepl/hacking_on_nrepl.html

### CODE REVIEW

File: src/clojure/nrepl/bencode.clj [hunk 1]
```diff
@@ -324,7 +324,7 @@
       :else (type thing))))
 
 (defmethod write-bencode :default
-  [output x]
```
Comment (bbatsov) [inline src/clojure/nrepl/bencode.clj:327]: I think you meant to have this as `_output` here.

File: src/clojure/nrepl/cmdline.clj [hunk 2]
```diff
@@ -373,7 +373,7 @@ Exit:      Control+D or (exit) or (quit)"
   "Connects to a running nREPL server and runs a REPL. Exits program when REPL
   is closed.
   Takes a map of nREPL CLI options."
-  [{:keys [host port transport] :as options}]
+  [{:keys [host port _transport] :as options}]
```
Comment (bbatsov) [inline src/clojure/nrepl/cmdline.clj:376]: Seems to me that in destructuring it's better to just remove redundant keys.

File: src/clojure/nrepl/util/completion.clj [hunk 3]
```diff
@@ -152,7 +152,7 @@
         (.endsWith path ".jar")
         (try (for [^JarEntry entry (enumeration-seq (.entries (JarFile. path)))]
                (.getName entry))
-             (catch Exception e))
+             (catch Exception e (println (.getMessage e))))
```
Comment (bbatsov) [inline src/clojure/nrepl/util/completion.clj:155]: Any particular reason you decided to print this error?

File: src/clojure/nrepl/util/lookup.clj [hunk 4]
```diff
@@ -8,6 +8,7 @@
   the API is subject to changes."
   {:author "Bozhidar Batsov"
    :added "0.8"}
+  #_:clj-kondo/ignore
```
Comment (bbatsov) [inline src/clojure/nrepl/util/lookup.clj:11]: What is this ignore for?

**Comment by bbatsov** [comment]
Thanks for the PR. I'll have to ponder on some of the changes, but overall it seems like a step in the right direction to me.

File: src/clojure/nrepl/util/lookup.clj (hunk 4 above)
Comment (bbatsov) [reply src/clojure/nrepl/util/lookup.clj:11]: Seems to me it's better to just update the test.

File: src/clojure/nrepl/cmdline.clj [hunk 5]
```diff
@@ -373,7 +373,7 @@ Exit:      Control+D or (exit) or (quit)"
   "Connects to a running nREPL server and runs a REPL. Exits program when REPL
   is closed.
   Takes a map of nREPL CLI options."
-  [{:keys [host port _transport] :as options}]
```
Comment (bbatsov) [inline src/clojure/nrepl/cmdline.clj]: Just to be clear - it'd be best to update all similar usages like this - just removed unused keys in destructuring.

File: src/clojure/nrepl/ack.clj [hunk 6]
```diff
@@ -35,7 +35,8 @@
       (h msg)
       (try
         (deliver @ack-port-promise port)
-        (t/send transport {:status :done})))))
+        (t/send transport {:status :done})
+        (catch Throwable e (println (.getMessage e)))))))
```
Comment (bbatsov) [inline src/clojure/nrepl/ack.clj:39]: I guess you can remove this `println` as well.

**Comment by solar05** [comment]
Okay, within a week do that :)

File: src/clojure/nrepl/cmdline.clj (hunk 2 above)
Comment (bbatsov) [reply src/clojure/nrepl/cmdline.clj:376]: I think you forgot to remove this `_transport`.

File: src/clojure/nrepl/middleware/print.clj [hunk 7]
```diff
@@ -101,7 +101,7 @@
   of the content written to that `PrintWriter` will be sent as messages on the
   transport of `msg`, keyed by `key`."
   ^java.io.PrintWriter
-  [key {:keys [transport] :as msg} {:keys [::buffer-size ::quota] :as opts}]
+  [key {:keys [transport] :as msg} {:keys [::buffer-size ::quota] :as _opts}]
```
Comment (bbatsov) [inline src/clojure/nrepl/middleware/print.clj:104]: You can remove this `:as`.

File: src/clojure/nrepl/middleware/print.clj [hunk 8]
```diff
@@ -136,9 +136,9 @@
   (transport/send transport (apply dissoc resp keys)))
 
 (defn- send-nonstreamed
-  [{:keys [transport] :as msg}
+  [{:keys [transport] :as _msg}
    resp
-   {:keys [::print-fn ::quota ::keys] :as opts}]
+   {:keys [::print-fn ::quota ::keys] :as _opts}]
```
Comment (bbatsov) [inline src/clojure/nrepl/middleware/print.clj:141]: And this one.

**Comment by bbatsov** [comment]
> Okay, within a week do that :)

Much appreciated! Thanks!

**Pull request merged**